.github/workflows/build-container.yml

```diff
@@ -199,6 +199,19 @@ jobs:
       - name: Display logs
         run: make logs OPTS=''
 
+      - name: Login to quay.io (on release/tag create)
+        if: startsWith(github.ref, 'refs/tags/v')
```

Comment:
If we only want to push containers in release create, I'd rather use that event, as it will happen AFTER release creation, while the push-tag event will happen in the early stage (BEFORE) of a release creation.

Reply:
@epuertat , as I understand this condition line, it checks if the current event is for a tag starting with v (when creating such), a new release will not trigger the images push action to the registry, as I believe is the desired behavior, right?

Reply:
Yes, but as I assume that you have created releases using the Github web, the first step is to create a tag, and you can even create a tag without a release (`release` is a Github concept, not a `git` one, as `tags`) or dismiss a release while creating it (but the tag will be already created).

Once you have created a tag, this CI will run and publish the container images, even if you haven't finished filling out the release doc or you decided to postpone the release. It's not a big difference, but IMHO it makes more sense that the containers are published AFTER the release is out, and not before.

Reply:
I based my expected workflow design on the manual push actions so far , actually there is one pre-release with two tags (v0.0.1 and v0.0.2), and those images were pushed on each new tag creation, right? 
The workflow has been changed as follows, let me know what you think:
- a simple commit push will not trigger the CI workflow anymore (@baum, please share your thoughts on that)
- a new tag that starts with with 'v' will trigger the CI workflow but the nvmeof images won't be pushed to registry.
- a new release (pre-release or published release) creation will also trigger the CI workflow and the nvmeof images will be pushed to registries.

Reply:
@epuertat @barakda 🖖
I'm understanding that in this scenario, once the CI workflow successfully completes, you're considering marking that specific codebase version as a release by tagging its corresponding SHA. Subsequently, you anticipate that the container images associated with this tagged SHA version will be pushed to a registry as part of the release process.

There are likely multiple ways to achieve this functionality. One potential approach is to introduce an additional workflow, possibly named 'Release,' which is triggered exclusively by the pushing of tags. This new workflow could encompass tasks such as code checkout, building, and pushing the container images.
```yaml
name: Release

on:
  push:
    tags:
      - '*'

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
    - name: Check out code
      ...
      
    - name: Build
      run: |
        echo "Building..."
        make build

    - name: Push
      run: |
        echo "Pushing..."
        make push

```
WDYT?

Reply:
Yeah, this could work too, but if you check examples of Github Actions, they tend to be single workflow and do the following:
```mermaid
graph LR
	B(Build) --> T1(Test 1) & T2(Test 2) & TN(Test N) --> R(Release) --> D(Deployment)
```

Regarding the triggering event, I think that "Release -> create" is the best one, as it happens after the **Publish release** button has been clicked, which clearly marks when a release is meant to be generally available (I guess it's acceptable to wait a few minutes until the containers are built and pushed to the registry):

![image](https://github.com/ceph/ceph-nvmeof/assets/37327689/1dea3a83-c10e-4b26-90d8-a649000c2955)